arch/risc-v/src/mpfs/mpfs_corespi.c

```diff
@@ -720,6 +731,52 @@ static int mpfs_spi_sem_waitdone(struct mpfs_spi_priv_s *priv)
   return nxsem_tickwait_uninterruptible(&priv->sem_isr, USEC2TICK(timeout));
 }
 
+/****************************************************************************
+ * Name: mpfs_spi_write_tx8
+ *
+ * Description:
+ *   Fill up the TX fifo with one word in configured size frames
+ *
+ * Input Parameters:
+ *   priv     - SPI private state data
+ *   nframes  - Number of 8-bit data per word
+ *   last     - true if this is the last word of the transfer
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static inline void mpfs_spi_write_tx8(struct mpfs_spi_priv_s *priv,
+                                      unsigned nframes,
+                                      bool last)
+{
+  uint8_t *data8 = (uint8_t *)priv->txbuf;
+  unsigned pos = priv->tx_pos++;
```

Comment:
Maybe this ++ can be simplified to nframes * frame_size ?

Reply:
nope, the priv->tx_pos counts the SPI words (priv->nbits size words), and this function just writes one of those. E.g. if priv->nbits == 16, but we have 8-bit corespi, this means that nframes == 2 (need to combine 2 8-bit spi frames to write one requested 16-bit frame one - this function writes 2 8-bit spi frames).

Reply:
Oh, or did you mean to just put this on one line? 

unsigned pos = nframes * priv->tx_pos++;

I felt that it makes it even more confusing....

Reply:
No, this is fine as is

Reply:
or, could keep the incrementing of priv->tx_pos in "mpfs_spi_load_tx_fifo", and just pass the current word index as an argument....